### DISCUSSION TITLE
Improved/Bundled Development API Keys

### DISCUSSION THREAD

**Discussion by ptlthg** [HypixelDev/PublicAPI#611] [Ideas]
**TLDR:** Please improve development keys. Either a lower request limit with longer expiry, or bundle dev keys in with production (or increased limit) applications. (I prefer a bundled key, or both)

While I generally think the new developer dashboard is a great new system, development keys only lasting 3 days before needing to be manually refreshed makes developing things a bit frustrating. It feels like each time I load up the project to put some work in this happens:

1. Hm, weird, requests are taking a while.
2. Oh, `Forbidden` errors?
3. *Realization*
4. I have to generate a new key and put it into my settings.

Yes, it takes only 30 seconds, but it's off-putting to need to jump through this hurdle so frequently.

I understand that this limitation exists to dissuade normal players from abusing these keys (like by using them in mods), but a better solution is possible. Providing longer lasting dev keys with a much worse request limit would be perfectly fine for me (and I suspect most developers). Something like 60 requests / 5 minutes would be perfect.

Alternatively, giving a key with no expiry but with a lower limit alongside approved production applications (or applications with a higher limit) would be great. 

I have been given an increased limit for my application, and whitelisted my server IP, so I can't exactly just use that one in development. Using this key would be a bad idea anyway, as less robust rate limiting implementations might not be reading the request headers to account for these extra requests. Providing another development key with these applications would make the developer experience way easier.
